lib/index.js

```diff
@@ -38,6 +42,35 @@ function initialize(config) {
   });
 }
 
+function filterPreviousUploads(uploadListPath, objects) {
+  var uploadedFileList;
+  if (uploadListPath) {
+    uploadedFileList = fs.readFileSync(uploadListPath, {encoding: 'utf8'});
+    uploadedFileList = uploadedFileList.split('\n');
+    if (!uploadedFileList[uploadedFileList.length-1]) uploadedFileList.pop();
+
+    uploadedFileList = _.map(uploadedFileList, function(anUploadedFile) {
+      return JSON.parse(anUploadedFile);
+    });
+
+    uploadedFilesById = _.groupBy(uploadedFileList, 'objectId');
+  }
+
+  objects = _.filter(objects, function(aFileObject) {
+    if (!aFileObject.fileName) return false; //null fileName causes error in indexOf later
```

Comment:
Is there somewhere in the script that checks if the file obj is valid? I think that would be its rightful place.  alternatively, is it possible for a file to exist but the name to not exist?  or is this a case of where the file is null and the method that constructed the list of Parse Files forgot to check that?  (I think it's probably the latter and that would make it the right place for this)